### PR TITLE
Add MongoDB connection pooling

### PR DESCRIPTION
**Pull request by jackvanb**
See https://github.com/ucla/lti-media-resources/pull/238 for details.

### CODE REVIEW

**Comment by rlorenzo** [comment]
@jackvanb I am getting this error when I run yarn dev and try to load the tool:

(node:21544) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'db' of undefined
    at Provider.<anonymous> (/Users/rex/Projects/lti-library-resources/src/server/index.js:43:32)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

But when I go back to the master branch I can load the LTI tool fine.

File: src/server/models/db.js [hunk 1]
```diff
@@ -9,7 +9,11 @@ module.exports.connect = async function (url) {
     return;
```
Comment (rlorenzo) [inline src/server/models/db.js:9]: @jackvanb This appears to be missing require('dotenv').config(); from here:

https://github.com/ucla/lti-media-resources/pull/238/files#diff-1497d44a565c9a4d1e06aaa06e751e9cd1c3e7b47ffa1d7829a6ae4e6405a11cR1

**Comment by rlorenzo** [comment]
@jackvanb Even with the latest changes it hangs at this step for me:

2021-02-25T02:47:43.150Z provider:namesAndRolesService Memberships retrieved

Nothing happens after that. Going back to master there is no hangup. Is this loading for you?

**Comment by jackvanb** [comment]
This now works for me in the production and development environment using the Atlas cluster. 

I also had to update my public key.

**Comment by rlorenzo** [comment]
Just noting, was also experiencing hanging, but that was because I didn't have this in my .env:

DB_MAX_POOL_SIZE=25
DB_MAX_IDLE_TIME_MS=60000

**Pull request merged**